promql/functions.go

```diff
@@ -1028,3 +1048,7 @@ func createLabelsForAbsentFunction(expr parser.Expr) labels.Labels {
 	}
 	return m
 }
+
+func stringFromArg(e parser.Expr) string {
```

Comment:
I would consider inlining this TBH. This function is super shallow and unsafe (casting), unless I am missing something,

Not a blocker.

Reply:
We do this unsafe casting even if we inline, so that is not an issue (I think that is safeguarded by the parser and we are expected to use it at right place). Moving out to a function make it concise and less error prone to get the string (since it can be now wrapped with `StepInvariantExpr` node and you can forget to unwrap it).